src/main/scala/com/yotpo/metorikku/input/kafka/KafkaInput.scala

```diff
@@ -1,12 +1,20 @@
 package com.yotpo.metorikku.input.kafka
 
+import java.util.Properties
+
 import com.yotpo.metorikku.input.Reader
-import org.apache.spark.sql.DataFrame
+import com.yotpo.metorikku.instrumentation.KafkaLagWriter
 import com.yotpo.metorikku.session.Session.getSparkSession
+import org.apache.kafka.clients.consumer.KafkaConsumer
+import org.apache.spark.sql.{DataFrame, SparkSession}
 
-case class KafkaInput(name: String, servers: Seq[String], topic: String,
+case class KafkaInput(name: String, servers: Seq[String], topic: String, consumerGroup: String,
                       options: Option[Map[String, String]]) extends Reader {
   def read(): DataFrame = {
+    val consumer = createKafkaConsumer
+    val lagWriter = new KafkaLagWriter(consumer)
+    val ss = SparkSession.builder().getOrCreate()
```

Comment:
Can we get the session from outside?

Reply:
Why? its a singleton object we can use anywhere? i hate passing the session around..